migration/src/main/java/com/intershop/customization/migration/common/MigrationPreparer.java

```diff
@@ -2,30 +2,75 @@
 
 import java.nio.file.Path;
 
+/**
+ * Interface for implementing migration preparers that handle specific migration tasks.
+ * </p>
+ * A migration preparer is responsible for transforming resources according to migration rules. Implementations should
+ * provide logic for modifying directories, files, and other resources to ensure compatibility with target platform
+ * versions.
+ */
 public interface MigrationPreparer
 {
     /**
-     * @param resource contains information, what needs to be migrated
+     * Migrates a resource.
+     *
+     * @param resource Path to the resource that needs to be migrated
      */
-    void migrate(Path resource);
+    @Deprecated(forRemoval = true)
```

Comment:
Usually a small documentation is then needed to explain, what to do for migration.